File: src (cpp)/animal/aves/colibri.cpp
#include "colibri.h"

Colibri::Colibri(int _x, int _y, int _weight) : def_ID(19),
                                                def_ratio_meat(0),
                                                def_ratio_plant(60) {
  ID = def_ID;
  position.SetX(_x);
  position.SetY(_y);
  ratio_meat = def_ratio_meat;
  ratio_plant = def_ratio_plant;
  is_land_animal = true;
  is_water_animal = false;
  is_air_animal = true;
  weight = _weight;
}
void Colibri::Interact() {
  std::cout << "The colibri is looking at you, apparently confused" << std::endl;
}

File: src/virtualzoo/animal/aves/Colibri.java
package virtualzoo.animal.aves;

import virtualzoo.animal.behavior.BehaviorTame;
import virtualzoo.animal.diet.Herbivora;

/** <p>
 * Kelas Colibri mendefinisikan atribut untuk colibri.
 * </p>
 * @author Holy Lovenia - 13515113
 * @version 3.0
 * @since 3.0
 */
public class Colibri extends Aves {

  /** <p>
   * Constructor
   *
   * Menciptakan colibri dengan posisi (locX,locY) dan berat animalWeight.
   * </p>
   * @param locX Nilai absis posisi.
   * @param locY Nilai ordinat posisi.
   * @param animalWeight Nilai berat colibri.
   */
  public Colibri(int locX, int locY, int animalWeight) {
    super();
    animalBehavior = new BehaviorTame();
    animalDiet = new Herbivora(animalWeight, 60);
    id = 18;
    position.setLocation(locX, locY);
    isLandAnimal = true;
    isWaterAnimal = false;
    isAirAnimal = true;
  }

  /** <p>
   * Mengatur interaksi colibri.
   * </p>
   */
  public void interact() {
    System.out.println("The colibri is looking at you, apparently"
        + "confused");
  }
}